CONTRIBUTING.md

```diff
@@ -15,39 +15,194 @@ Otherwise, go read Docker's
+
+(Throughout this document, `<username>` refers to your Github username.)
+
+#### Clone Swarm and set remotes
+
+Now clone the main Swarm repo:
+
 ```sh
-go get -d github.com/docker/swarm
+mkdir -p $GOPATH/src/github.com/docker
+cd $GOPATH/src/github.com/docker
+git clone https://github.com/docker/swarm.git
+cd swarm
 ```
 
-#### Start hacking
+For the easiest contribution workflow, we will set up remotes as follows:
+  * `origin` is your personal fork
+  * `upstream` is `docker/swarm`
+
+Run these commands:
 
-First, prepare dependencies for `swarm` and try to compile it.
 ```sh
-cd src/github.com/docker/swarm
-$GOBIN/godep restore
-go test -v -race ./...
-go install
+git remote remove origin
+git remote add origin https://github.com/<username>/swarm.git
+git remote add upstream https://github.com/docker/swarm.git
+git remote set-url --push upstream no-pushing
 ```
 
-#### Adding New Dependencies
+You can check your configuration like this:
+```sh
+$ git remote -v
+origin     https://github.com/mgoelzer/swarm.git (fetch)
+origin     https://github.com/mgoelzer/swarm.git (push)
+upstream   https://github.com/docker/swarm.git (fetch)
+upstream   no-pushing (push)
+```
+
+#### Pull `docker/swarm` changes into your fork
+
+As the `docker/swarm` moves forward, it will accumulate commits that are not in your personal fork.  To keep your fork up to date, you need to periodically rebase:
+
+```sh
+git fetch upstream master
+git rebase upstream/master
+git push -f origin master
+```
+
+Don't worry:  fetching the upstream and rebasing will not overwrite your local changes.
+
+### Build the Swarm binary
+
+Build the binary, installing it to `$GOPATH/bin/swarm`:
+
+```sh
+cd $GOPATH/src/github.com/docker/swarm
+godep go install .
+```
+ 
+Run the binary you just created:
+
+```sh
+$GOPATH/bin/swarm help
+$GOPATH/bin/swarm create
+$GOPATH/bin/swarm manage token://<cluster_id>
+$GOPATH/bin/swarm join --addr=<node_ip:2375> token://<cluster_id>
+```
+
+#### Create a Swarm image for deployment
+
+Swarm is distributed as a Docker image with a single `swarm` binary inside.  To create the image:
+
+```sh
+docker build -t my_swarm .
+```
+
+Now you can run the same commands as above using the container:
+
+```sh
+docker run my_swarm help
+docker run -d my_swarm join --addr=<node_ip:2375> token://<cluster_id>
+docker run -d -p <manager_port>:2375 my_swarm manage token://<cluster_id>
+```
+
+For complete documentation on how to use Swarm, refer to the Swarm section of [docs.docker.com](http://docs.docker.com/).
+
+
+### Run tests
+
+To run unit tests:
+
+```sh
+godep go test -race ./...
+```
+
+To run integration tests:
+
+```sh
+./test/integration/run.sh
+```
+
+You can use this command to check if `*.bats` files are formatted correctly; otherwise your PR fail docker/dco-signed.
```

Comment:
It's not related to the DCO, we shouldn't mention it here.
Also, since we advise them to look at the format of the bats file, we should talk about golint here, to check the format of their go files